timemachine/cpp/src/nonbonded_interaction_group.cu

```diff
@@ -113,11 +107,7 @@ NonbondedInteractionGroup<RealType>::NonbondedInteractionGroup(
 
     gpuErrchk(cudaPeekAtLastError());
     cudaSafeMalloc(&d_sort_storage_, d_sort_storage_bytes_);
-    // We will sort so that the row atoms are always first for the nblist. Cheaper to set once than to
```

Comment:
This bit here is what was biting me. I was passing `d_row_atom_idxs_` directly to the nblist which is not what the ixn group is set to do. 

Now calling `set_atom_indices_device` will do the right thing for any set of row/column indices.